spec/forms/addresses/care_of_form_spec.rb

```diff
@@ -0,0 +1,213 @@
+require "rails_helper"
+
+RSpec.describe Addresses::CareOfForm, type: :form do
+  subject(:instance) { described_class.new(params) }
+
+  let(:address) { create(:address, location: "correspondence") }
+  let(:care_of) { "no" }
+
+  let(:params) do
+    {
+      model: address,
+      care_of:,
+      care_of_first_name:,
+      care_of_last_name:,
+      care_of_organisation_name:,
+    }
+  end
+
+  describe "#save" do
+    subject(:call_save) { instance.save }
+
+    before { call_save }
+
+    context "with care of person chosen" do
+      let(:care_of) { "person" }
+      let(:care_of_last_name) { "Smith" }
+      let(:care_of_first_name) { "Bob" }
+      let(:care_of_organisation_name) { nil }
+
+      it "updates the address care_of fields" do
+        expect(address.care_of).to eq "person"
+        expect(address.care_of_last_name).to eq "Smith"
+        expect(address.care_of_first_name).to eq "Bob"
+        expect(address.care_of_organisation_name).to be_nil
+      end
+    end
+
+    context "with care of organisation chosen" do
+      let(:care_of) { "organisation" }
+      let(:care_of_last_name) { nil }
+      let(:care_of_first_name) { nil }
+      let(:care_of_organisation_name) { "An Organisation Name" }
+
+      it "updates the address care_of fields" do
+        expect(address.care_of).to eq "organisation"
+        expect(address.care_of_last_name).to be_nil
+        expect(address.care_of_first_name).to be_nil
+        expect(address.care_of_organisation_name).to eq "An Organisation Name"
+      end
+    end
+
+    context "with no chosen" do
+      let(:care_of) { "no" }
+      let(:care_of_last_name) { nil }
+      let(:care_of_first_name) { nil }
+      let(:care_of_organisation_name) { nil }
+
+      it "updates the address care_of fields" do
+        expect(address.care_of).to eq "no"
+        expect(address.care_of_last_name).to be_nil
+        expect(address.care_of_first_name).to be_nil
+        expect(address.care_of_organisation_name).to be_nil
+      end
+    end
+
+    context "with no answer chosen" do
+      let(:care_of) { nil }
+      let(:care_of_last_name) { nil }
+      let(:care_of_first_name) { nil }
+      let(:care_of_organisation_name) { nil }
+
+      it "is invalid" do
+        expect(instance).not_to be_valid
+      end
+
+      it "adds custom blank error message" do
+        error_messages = instance.errors.messages.values.flatten
+        expect(error_messages).to include("Select if you want to add a 'care of' recipient for your client's email")
+      end
+    end
+
+    context "with missing person name" do
+      let(:care_of) { "person" }
+      let(:care_of_last_name) { "Smith" }
+      let(:care_of_first_name) { nil }
+      let(:care_of_organisation_name) { nil }
+
+      it "is invalid" do
+        expect(instance).not_to be_valid
```

Comment:
The test description doesn't match the expectation invalid or valid

Reply:
I think this description does match(is invalid not_to be_valid), or is it not clear?